docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,244 @@
+  $(BRAND)/$(PRODUCT)/$(DEVICE)/$(BOARD):$(VERSION.RELEASE)/$(ID)/$(VERSION.INCREMENTAL):$(TYPE)/\$(TAGS)
+
+  - `"google/sdk_gphone_x86/generic_x86:9/PSR1.180720.075/5124027:user/release-keys"`
+
+- `Device.osBuildId: string`
+
+  A string that identifies the build number of the operating system.
+
+  - `"MMB29K"`
+
+## Methods
+
+### `Device.getPlatformFeaturesAsync()` (Android Only)
+
+Get a list of features that are available on the system.
+
+#### Returns
+
+A Promise that resolves to an array of strings, each containing the name of the available feature on the current device system.
+
+**Examples**
+
+```js
+await Device.getPlatformFeaturesAsync().then(allFeatures => {
+  // Array [
+  // "android.software.adoptable_storage",
+  // "android.hardware.sensor.accelerometer",
+  // "android.software.backup",
+  // "android.hardware.touchscreen"
+  // ]
+});
+```
+
+### `Device.hasPlatformFeatureAsync(feature)` (Android Only)
+
+Tells if the device has a specific system feature. Can get all available system features in `Device.getSystemFeatureAsync()`.
+
+#### Arguments
+
+- **feature (_string_)** -- A string of the feature we want to know that the device has.
+
+#### Returns
+
+Returns a `Promise<boolean>` that resolves the `boolean` value for whether the device has the system feature passed to the function.
+
+**Examples**
+
+```js
+await Device.hasPlatformFeatureAsync('amazon.hardware.fire_tv'); // true or false
+```
+
+### `Device.getMaxMemoryAsync()` (Android Only)
+
+Returns the maximum amount of memory that the Java virtual machine will attempt to use. If there is no inherent limit then the value [Long.MAX_VALUE](https://developer.android.com/reference/java/lang/Long.html#MAX_VALUE) will be returned.
+
+#### Returns
+
+Returns a Promise that resolves the maximum available memory that the Java vm will use, in bytes.
+
+**Examples**
+
+```js
+Device.getMaxMemoryAsync().then(maxMemory => {
+  // 402653184
+});
+```
+
+### `Device.isSideLoadingEnabled()` (Android Only)
+
+Whether applications can be installed for this user via the system's [Intent#ACTION_INSTALL_PACKAGE](https://developer.android.com/reference/android/content/Intent.html#ACTION_INSTALL_PACKAGE) mechanism.
+
+#### Returns
+
+Returns a Promise that resolves to a boolean that represents whether the calling package is allowed to request package installation.
+
+**Examples**
+
+```js
+Device.isSideLoadingEnabled().then({
+  // true or false
+});
+```
+
+### `Device.getUptimeAsync()`
+
+Gets the uptime since the last reboot of the device, in milliseconds.
+
+#### Returns
+
+Returns a promise that resolves to a number that represents the milliseconds since last reboot, not counting time spent in deep sleep.
+
+**Examples**
+
+```js
+Device.getUptimeAsync().then(uptime => {
+  // 4371054
+});
+```
+
+### `Device.isRootedExperimentalAsync()` (Android only)
+
+Checks whether the device has been rooted. It is not surely reliable because it's hard to test all rooted devices thoroughly.
```

Comment:
```suggestion
Checks whether the device has been rooted. It is not completely reliable because it's hard to test all rooted devices thoroughly.
```

> because it's hard to test all rooted devices thoroughly.

I don't understand what this means and what the specific challenges are. Think about this from the perspective of a finance app trying to prevent fraud and explain the threat model to them.